SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetConnection.cpp

```diff
@@ -230,3 +206,38 @@ void USpatialNetConnection::OnHeartbeat()
 {
 	SetHeartbeatTimeoutTimer();
 }
+
+void USpatialNetConnection::ClientNotifyClientHasQuit()
+{
+	if (PlayerControllerEntity != SpatialConstants::INVALID_ENTITY_ID)
+	{
+		if (!Cast<USpatialNetDriver>(Driver)->StaticComponentView->HasAuthority(PlayerControllerEntity,
+																				SpatialConstants::HEARTBEAT_COMPONENT_ID))
+		{
+			UE_LOG(LogSpatialNetConnection, Warning,
+				   TEXT("Quit the game but no authority over Heartbeat component: NetConnection %s, PlayerController entity %lld"),
+				   *GetName(), PlayerControllerEntity);
+			return;
+		}
+
+		FWorkerComponentUpdate Update = {};
+		Update.component_id = SpatialConstants::HEARTBEAT_COMPONENT_ID;
+		Update.schema_type = Schema_CreateComponentUpdate();
+		Schema_Object* ComponentObject = Schema_GetComponentUpdateFields(Update.schema_type);
+
+		Schema_AddBool(ComponentObject, SpatialConstants::HEARTBEAT_CLIENT_HAS_QUIT_ID, true);
+
+		Cast<USpatialNetDriver>(Driver)->Connection->SendComponentUpdate(PlayerControllerEntity, &Update);
+	}
+	else
+	{
+		UE_LOG(LogSpatialNetConnection, Warning, TEXT("Quitting before Heartbeat component has been initialized: NetConnection %s"),
+			   *GetName());
+	}
+}
+
+void USpatialNetConnection::OnControllerDestroyed(AActor* DestroyedActor)
```

Comment:
Nit - can remove the argument name. This might also give warnings on some compilers.